stat/distuv/poisson.go

```diff
@@ -0,0 +1,113 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+	"math/rand"
+
+	"gonum.org/v1/gonum/mathext"
+)
+
+// Poisson implements the Poisson distribution, a discrete probability distribution
+// that expresses the probability of a given number of events occurring in a fixed
+// interval of time and/or space.
+// The poisson distribution has density function:
+//  f(k) = λ^k / k! e^(-λ)
+// For more information, see https://en.wikipedia.org/wiki/Poisson_distribution.
+type Poisson struct {
+	// Lambda is the average number of events in an interval.
+	// Lambda must be greater than 0.
+	Lambda float64
+
+	Source *rand.Rand
+}
+
+// CDF computes the value of the cumulative distribution function at x.
+func (p Poisson) CDF(x float64) float64 {
+	if x < 0 {
+		return 0
+	}
+	fx1 := math.Floor(x + 1)
+	return mathext.GammaIncComp(fx1, p.Lambda)
```

Comment:
You can put this expression in-line `return mathext.GammaIncComp(math.Floor(x + 1), p.Lambda)`.

Also, maybe it's just me (very likely), but this is confusing in the context of the WP article since it gives this as \Gamma(\floor{k+1}, \lambda) / \floor{k}!.